tensorflow/python/ops/variables.py

```diff
@@ -226,7 +226,12 @@ def __init__(self,
           expected_shape=expected_shape)
 
   def __str__(self):
-    return str(self._snapshot)
+    return "<Variable name='%s' shape=%s dtype=%s>" % (
+            self.name(), self.get_shape(), self.dtype().name)
```

Comment:
Done, except for the second part. The reason I did `self.dtype().name` is because of this method in Variable: https://github.com/iRapha/tensorflow/blob/ab88819372cd83cb3db7fa5419946e3e4846ba29/tensorflow/python/ops/variables.py#L745